app/components/polls/polls-edit/polls-edit-directive.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+angular.module('polls.edit.directive', [
+  'apiServices.poll.factory',
+  'BettyCropper',
+  'lodash',
+  'saveButton.directive',
+  'topBar'
+]).constant('RESPONSE_TYPES', [
+  {
+    name: 'Text Only',
+    value: 'Text'
+  },
+  {
+    name: 'Image + Text',
+    value: 'Image'
+  }
+])
+  .directive('pollsEdit', function (routes) {
+    return {
+      templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
+      controller: function (_, $location, $q, $routeParams, $scope, $window, Poll) {
+        // populate model for use
+        if ($routeParams.id === 'new') {
+          $scope.model = Poll.$build();
+          $scope.isNew = true;
+        } else {
+          $scope.model = Poll.$find($routeParams.id);
+        }
+
+        window.onbeforeunload = function (e) {
```

Comment:
I wrote in these patterns a while ago for special coverage, where this is probably coming from, and we might as well improve on it while we're at it.

In this instance, if there's no other clashing `onbeforeunload` event handlers, we should be able to make the event handler here and the `$destroy` thing below look like this:

```js
var removeUnsavedChangesGuard = $window.$on('beforeunload', function (e) { ... });
$scope.$on('$destroy', removeUnsavedChangesGuard);
```